rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/DefaultWorkspaceService.java

```diff
@@ -72,13 +84,24 @@ public void initialize(ClientCapabilities clientCap, @Nullable List<WorkspaceFol
 
         var clientWorkspaceCap = clientCap.getWorkspace();
 
-        if (clientWorkspaceCap != null && Boolean.TRUE.equals(clientWorkspaceCap.getWorkspaceFolders())) {
-            var workspaceCap = new WorkspaceFoldersOptions();
-            workspaceCap.setSupported(true);
-            workspaceCap.setChangeNotifications(true);
-            capabilities.setWorkspace(new WorkspaceServerCapabilities(workspaceCap));
+        WorkspaceServerCapabilities workspaceCapabilities = new WorkspaceServerCapabilities();
+        if (clientWorkspaceCap != null) {
+            if (clientWorkspaceCap.getWorkspaceFolders()) {
+                var folderOptions = new WorkspaceFoldersOptions();
+                folderOptions.setSupported(true);
+                folderOptions.setChangeNotifications(true);
+                workspaceCapabilities.setWorkspaceFolders(folderOptions);
+            }
+
+            if (clientWorkspaceCap.getFileOperations().getDidRename()) {
+                var fileOperationCapabilities = new FileOperationsServerCapabilities();
+                fileOperationCapabilities.setDidRename(new FileOperationOptions(
+                    List.of(new FileOperationFilter(new FileOperationPattern("**")))));
```

Comment:
this now let's rascal also get `didRename` events for all files, while it should only receive them for `.rsc` files. can we make that pattern a parameter for the  `DefaultWorkspaceService`?

Reply:
This code has moved back to `RascalWorkspaceService`. If we want to improve on this it should be done in a separate PR.